client/post-editor/editor-featured-image/index.jsx

```diff
@@ -1,58 +1,68 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component } from 'react';
 import classnames from 'classnames';
 import Gridicon from 'gridicons';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
 import MediaModal from 'post-editor/media-modal';
+import MediaActions from 'lib/media/actions';
 import PostActions from 'lib/posts/actions';
 import PostUtils from 'lib/posts/utils';
 import * as stats from 'lib/posts/stats';
 import EditorFeaturedImagePreviewContainer from './preview-container';
 import Button from 'components/button';
+import { getMediaItem } from 'state/selectors';
+import { getFeaturedImageId } from 'lib/posts/utils';
+import QueryMedia from 'components/data/query-media';
+import { localize } from 'i18n-calypso';
 
-export default React.createClass( {
-	displayName: 'EditorFeaturedImage',
-
-	propTypes: {
+class EditorFeaturedImage extends Component {
+	static propTypes = {
 		maxWidth: React.PropTypes.number,
 		site: React.PropTypes.object,
 		post: React.PropTypes.object,
 		selecting: React.PropTypes.bool,
-		onImageSelected: React.PropTypes.func
-	},
-
-	getDefaultProps() {
-		return {
-			maxWidth: 450,
-			onImageSelected: () => {}
-		};
-	},
-
-	getInitialState() {
-		return {
+		onImageSelected: React.PropTypes.func,
+		featuredImage: React.PropTypes.object,
+	};
+
+	static defaultProps = {
+		maxWidth: 450,
+		onImageSelected: () => {}
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
 			isSelecting: false
 		};
-	},
+	}
+
+	showMediaModal = () => {
+		const { featuredImage, site } = this.props;
+
+		if ( featuredImage ) {
+			MediaActions.setLibrarySelectedItems( site.ID, [ featuredImage ] );
```

Comment:
`[ featuredImage ]` is necessary because `SET_MEDIA_LIBRARY_SELECTED_ITEMS` [expects an array.](https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/library-selected-store.js#L99)